### PR TITLE
Update vllm rule for better EP performance

### PR DESCRIPTION
**Pull request by NuojCheng**
# Description

It refactor the vllm logical rule and enable attn dp expert for optimal inference performance.

This PR also updates the function names from `create_nnx_model` to `from_pretrained` following updates from #3450 .

# Tests

Vllm test: https://paste.googleplex.com/4812933185011712

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

**Review by NicoGrande — APPROVED** [review]
LGTM!

File: src/maxtext/integration/vllm/maxtext_vllm_adapter/adapter.py [hunk 1]
```diff
@@ -254,7 +254,7 @@ def load_weights(self, rng_key: jax.Array) -> None:
       return
 
     with self.mesh, nn.logical_axis_rules(self.maxtext_config.logical_axis_rules):
-      model, _ = model_creation_utils.create_nnx_model(
+      model = model_creation_utils.from_pretrained(
```
Comment (gobbleturk) [inline src/maxtext/integration/vllm/maxtext_vllm_adapter/adapter.py:257]: why is this part of the PR? Is this some artifact of rebasing?

Comment (NuojCheng) [reply src/maxtext/integration/vllm/maxtext_vllm_adapter/adapter.py:257]: not from rebasing but vllm decoding fails at head now. Put in the same PR for decoding test purpose.

Comment (bvandermoon) [reply src/maxtext/integration/vllm/maxtext_vllm_adapter/adapter.py:257]: Mind adding this info to the PR description?

Comment (NuojCheng) [reply src/maxtext/integration/vllm/maxtext_vllm_adapter/adapter.py:257]: > Mind adding this info to the PR description?

Done

**Review by bvandermoon — APPROVED** [review]
